### PR TITLE
[Core][nightly test] add dataset shuffle data loader test

### PR DESCRIPTION
**Pull request by scv119**
This adds tests for dataset shuffle data loader (https://github.com/ray-project/ray/issues/17553).

running: https://beta.anyscale.com/o/anyscale-internal/projects/prj_SVFGM5yBqK6DHCfLtRMryXHM/clusters/ses_mfZFWrXQ8X681qTEVfewUZFe

### CODE REVIEW

File: release/nightly_tests/nightly_tests.yaml [hunk 1]
```diff
@@ -315,3 +315,12 @@
...
     prepare: python wait_cluster.py 32 1000
     script: python dask_on_ray/dask_on_ray_sort.py --nbytes 1_000_000_000_000 --npartitions 1000 --num-nodes 31 --ray --data-dir /tmp/ray --s3-bucket core-nightly-test
+
+- name: dataset_shuffle_data_loader
+  cluster:
+    app_config: shuffle_data_loader/shuffle_data_loader_app_config.yaml
+    compute_template: shuffle_data_loader/shuffle_data_loader_compute.yaml
+
+  run:
+    timeout: 7200
```
Comment (rkooo567) [inline release/nightly_tests/nightly_tests.yaml:325]: How long does it take now?

**Review by rkooo567 — APPROVED** [review]

**Review by rkooo567 — COMMENTED** [review]
Btw, can you copy the config & add num_gpus=1 to consume task?  like this PR https://github.com/ray-project/ray/pull/17773/files? That's more similar setup to @richardliaw's.

**Comment by rkooo567** [comment]
https://github.com/ray-project/ray/pull/17773/files will be closed in favor of this PR

File: release/nightly_tests/nightly_tests.yaml (hunk 1 above)
Comment (scv119) [reply release/nightly_tests/nightly_tests.yaml]: huh it runs much slower with gpus as we have less gpus available...

Comment (rkooo567) [reply release/nightly_tests/nightly_tests.yaml]: Hmm I see. Maybe we can make it run for an hour instead and fail it like my script?

Comment (scv119) [reply release/nightly_tests/nightly_tests.yaml]: nvm, reduced the number of workers and repeats:
https://beta.anyscale.com/projects/prj_SVFGM5yBqK6DHCfLtRMryXHM/sessions/ses_cGNFUqzRqZAr98ZMWLb5i2g8

**Comment by scv119** [comment]
this is good to be merged.

**Pull request merged**